### PR TITLE
Issue #654: Add support for python 3.11

### LINKED ISSUE #654: Add support for Python 3.11
Probably git-machete will work under Python 3.11 out of the box once released (~Oct 24, 2022).
Still, let's add sections in tox and CI config, as for the other supported versions.

### PR DESCRIPTION
**Pull request by amalota**

### CODE REVIEW

**Comment by amalota** [comment]
@PawelLipski We need to postpone this PR at least for couple days as it looks like the final release of python 3.11 will take place on 24.10.22 per [official python website](https://www.python.org/downloads/release/python-3110rc2/)

File: .circleci/config.yml [hunk 1]
```diff
@@ -98,6 +98,19 @@ jobs:
...
+    steps:
+      - checkout
+      - run:
+          name: Run tests against Python 3.11 and git 2.38.1 (with building docs)
+          command: PYTHON_VERSION=3.11 GIT_VERSION=2.38.1 BUILD_SPHINX_DOCS=true CHECK_PY_DOCS_UP_TO_DATE=true bash ci/tox/ci-run.sh
+      - store_test_results:
+          path: test-results/
+      - store_artifacts:
+          path: docs/html/
+          destination: docs
```
Comment (PawelLipski) [inline .circleci/config.yml]: Remove

File: .circleci/config.yml [hunk 2]
```diff
@@ -98,6 +98,19 @@ jobs:
           path: docs/html/
           destination: docs
 
+  "python 3_11 git 2_38_1":
+    executor: ubuntu_executor
+    steps:
+      - checkout
+      - run:
+          name: Run tests against Python 3.11 and git 2.38.1 (with building docs)
+          command: PYTHON_VERSION=3.11 GIT_VERSION=2.38.1 BUILD_SPHINX_DOCS=true CHECK_PY_DOCS_UP_TO_DATE=true bash ci/tox/ci-run.sh
```
Comment (PawelLipski) [inline .circleci/config.yml]: Plus remove flags for docs as well

File: .circleci/config.yml [hunk 3]
```diff
@@ -98,6 +98,19 @@ jobs:
           path: docs/html/
           destination: docs
 
+  "python 3_11 git 2_38_1":
+    executor: ubuntu_executor
+    steps:
+      - checkout
+      - run:
+          name: Run tests against Python 3.11 and git 2.38.1 (with building docs)
```
Comment (PawelLipski) [inline .circleci/config.yml]:
```suggestion
          name: Run tests against Python 3.11 and git 2.38.1
```

**Comment by PawelLipski** [comment]
Python 3.11 is there!

**Comment by PawelLipski** [comment]
Pls add `ci/circleci: python 3_11 git 2_38_1` to required checks for both `develop` and `master` once this is merged (I've restored your Admin)

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**